Partition (WBP)/Functions/VI0034C.VTI0034.fun.yml

```diff
@@ -0,0 +1,256 @@
+    #Com_Owner.mthIssue_Runtime_Message( #this_rc )
+    #Com_Owner.mthAdd_SubMsg( ER ('- Unexpected ' + #this_rc) ('Deleting on record ' + #$$@#.AsString) )
+    #this_rc := ER
+    Leave
+    Endif
+    Endselect
+    
+    If (#this_rc = OK)
+    #Com_Owner.mthAdd_SubMsg( OK ('- Successful') '' )
+    Endif
+    Endif
+    Else
+    #Com_Owner.mthAdd_SubMsg( ER ('- Unexpected ' + #this_rc) '' )
+    #this_rc := ER
+    Endif
+    
+    Clr_List Named(#w_lOriginal)
+    Endroutine
+    
+    *
+    *
+    *
+    Mthroutine Name(mthInsert)
+    Define_Map For(*BOTH) Class(#BIFA002A) Name(#this_rc)
+    
+    * Create Test Data list
+    
+    * Fields: #$$@# #@#$ #@@@@@@@@@
+    
+    #$$@# := 11111
+    #@#$ := 1111111
+    #@@@@@@@@@ := 111111
+    Add_Entry To_List(#w_lOriginal)
+    
+    #$$@# := 2222
+    #@#$ := 222222
+    #@@@@@@@@@ := 2222222
+    Add_Entry To_List(#w_lOriginal)
+    
+    #$$@# := 333
+    #@#$ := 333333
+    #@@@@@@@@@ := 2333333
+    Add_Entry To_List(#w_lOriginal)
+    
+    #$$@# := 4444444
+    #@#$ := 444433
+    #@@@@@@@@@ := 4444444
+    Add_Entry To_List(#w_lOriginal)
+    
+    #$$@# := 555555
+    #@#$ := 5555
+    #@@@@@@@@@ := 55
+    Add_Entry To_List(#w_lOriginal)
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('Insert: ' + #LF_Original_Count.AsString + ' record(s)') '' )
+    
+    Selectlist Named(#w_lOriginal)
+    Insert Fields(*ALL) To_File(@@#$) Io_Status(#this_rc) Io_Error(*NEXT) Val_Error(*NEXT)
+    
+    If (#this_rc <> OK)
+    #Com_Owner.mthIssue_Runtime_Message( #this_rc )
+    #Com_Owner.mthAdd_SubMsg( ER ('- Unexpected ' + #this_rc) ('Inserting record ' + #$$@#.AsString) )
+    Leave
+    Endif
+    Endselect
+    
+    If (#this_rc = OK)
+    #Com_Owner.mthAdd_SubMsg( OK ('- Successful') '' )
+    Endif
+    Endroutine
+    
+    *
+    * mthUpdate
+    *
+    Mthroutine Name(mthUpdate)
+    Define_Map For(*BOTH) Class(#BIFA002A) Name(#this_rc)
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('Update: ' + #LF_Original_Count.AsString + ' record(s)') '' )
+    
+    Select Fields(*ALL) From_File(@@#$)
+    #@#$ := 911119
+    Update Fields(#@#$) In_File(@@#$) Io_Status(#this_rc) Io_Error(*NEXT) Val_Error(*NEXT)
+    
+    If (#this_rc <> OK)
+    #Com_Owner.mthIssue_Runtime_Message( #this_rc )
+    #Com_Owner.mthAdd_SubMsg( ER ('- Unexpected ' + #this_rc) ('Updating record ' + #$$@#.AsString) )
+    #this_rc := ER
+    Leave
+    Endif
+    Endselect
+    
+    If (#this_rc = OK)
+    #Com_Owner.mthAdd_SubMsg( OK ('- Successful') '' )
+    Endif
+    Endroutine
+    
+    *
+    * mthSelect
+    *
+    Mthroutine Name(mthSelect)
+    Define_Map For(*BOTH) Class(#BIFA002A) Name(#this_rc)
+    
+    #Com_Owner.mthAdd_SubMsg( OK ('Test the updated : ' + #LF_Original_Count.AsString + ' record(s)') '' )
+    
+    #LF_@#$ := 911119
+    Select Fields(*ALL) From_File(@@#$)
+    
+    * Validate all the rows and only #$$@# field has been updated using mthUpdate method
+    
+    If (#LF_@#$ *NE #@#$)
```

Comment:
This should only be tested if OK.
And all Fields should be tested.

Reply:
Will resolve it as per comment

Reply:
Tested based on the 'OK' flag status